assets/js/testScript.js

```diff
@@ -0,0 +1,488 @@
+        background: "url(./assets/images/congratulations.png)",
+        width: "830px",
+        height: "80px",
+        position: "absolute",
+        left: "400px",
+        top: "50px"
+      })
+
+    }else if (gameEnd === true && playerHealth < 0){
+      $('.landingScreen').show()
+      $(".maplestoryImage").css({
+        background: "url(./assets/images/gameOver.png)",
+        width: "570px",
+        height: "80px",
+        position: "absolute",
+        left: "520px",
+        top: "50px"
+    })
+  }
+  }, 1000)
+
+
+  setInterval(function () {                 // check whether boss is to be spawned, if yes set image to visible and spawn boss
+    if (bossLevel) {
+      $bossSign.css('visibility', 'visible')
+      if (mobsOnScreen < 1 && bossDead === false) { generateBoss() } // ensure no other bosses are on screen prior to spawning
+    }
+  }, 1000)
+
+  function restartGame () {    // function to reset all fields when game restarts
+    $('.landingScreen').hide()
+    playerHealth = 1000
+    $hpBar.text(`${playerHealth}/1000`)
+    $bossSign.css('visibility', 'hidden')
+    $hpBar.css('width', `200px`)
+    mana = 202
+    $mpBar.text(`${mana}/202`)
+    $mpBar.css('width', `202px`)
+    mobsOnScreen = 0
+    gameEnd = false
+    bossLevel = false
+    levelOneMobCount = 10
+    bossDead = false
+    runSpawner()
+    for (key in mobArray) {
+      mobArray[key].jTarget.remove()
+    }
+    mobArray = []
+    for (key in bossArray) {
+      bossArray[key].jTarget.remove()
+    }
+    bossArray = []
+  }
+
+  function runSpawner () {  // spawn mobs based on limit set per level
+    var mobSpawn = setInterval(function () {
+      if (mobsOnScreen < levelOneMobCount) {
+        generateMob()
+      }
+    }, 3000)
+  }
+
+  function generateMob () {     //run mob spawn class
+    var id = Math.floor(Math.random() * 1000)
+    mobArray[id] = new Mob(id)
+    mobArray[id].addChar()
+    mobsOnScreen += 1
+  }
+  function generateBoss () {      // run boss spawn class
+    var id = Math.floor(Math.random() * 1000)
+    bossArray[id] = new Boss(id)
+    bossArray[id].addBoss()
+    mobsOnScreen += 1
+  }
+
+  var manaRegen = setInterval(function () {  //mana regeneration function
+    if (mana < 200) {
+      mana += 4
+      $mpBar.text(`${mana}/200`)
+      $mpBar.css('width', `${$mpBar.width() + 4}px`)
+    }
+  }, 1000)
+
+  setInterval(function () {       // collision check for bullets and player contact against mobs
+    for (key in mobArray) {
+      bulletCollisionCheck($bullet, mobArray[key])
+      playerMobCollisionCheck($player, mobArray[key])
+    }
+  }, 30)
+
+  setInterval(function () {       // collision check for bullets and player contact against boss
+    for (key in bossArray) {
+      bulletCollisionCheck($bullet, bossArray[key])
+      playerMobCollisionCheck($player, bossArray[key])
+    }
+  }, 30)
+
+  $body.on('keydown', function (e) {  //player keyboard press detection
+    var $player = $('.player')
```

Comment:
already defined at 122

Reply:
yeah was used for the checking portion when i was running the code, again i forgot to remove :(